exonum/src/proto/binary_map.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::proto;
+use exonum_crypto::Hash;
+use exonum_merkledb::{BinaryValue, ObjectHash};
+use exonum_proto::ProtobufConvert;
+use protobuf::Message;
+
+use std::{borrow::Cow, cmp::Ordering, collections::BTreeMap, iter::FromIterator};
+
+/// Protobuf wrapper type to store small maps of non-scalar keys and values.
+/// Stored keys are ordered and duplicate keys are forbidden.
+#[derive(Debug, Default, Clone, Eq, PartialEq, Ord, PartialOrd)]
+#[derive(Serialize, Deserialize)]
+pub struct BinaryMap<K: Ord, V>(pub BTreeMap<K, V>);
```

Comment:
Nit: What about changing naming to something like `OrderedMap`? `BinaryMap` looks slightly confusing (what exactly is binary?).